instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb

```diff
@@ -0,0 +1,107 @@
+          def endpoint_render(_name, start, _finish, _id, payload)
+            span = OpenTelemetry::Instrumentation::Rack.current_span
+            return unless span.recording?
+
+            span.add_event('grape.endpoint_render', timestamp: start)
+          end
+
+          # Handles the endpoint_run_filters events, recording them as a span event
+          def endpoint_run_filters(_name, start, finish, _id, payload)
+            span = OpenTelemetry::Instrumentation::Rack.current_span
+            return unless span.recording?
+
+            filters = payload[:filters]
+            type = payload[:type]
+
+            # Prevent submitting empty filters
+            return if (!filters || filters.empty?) || !type || (finish - start).zero?
+
+            attributes = { 'grape.filter.type' => type.to_s }
+            span.add_event('grape.endpoint_run_filters', attributes: attributes, timestamp: start)
+          end
+
+          # Handles the format_response event, recording it as a span event
+          def format_response(_name, start, _finish, _id, payload)
+            span = OpenTelemetry::Instrumentation::Rack.current_span
+            return unless span.recording?
+
+            attributes = {
+              'grape.formatter.type' => formatter_type(payload[:formatter])
+            }
+            span.add_event('grape.format_response', attributes: attributes, timestamp: start)
+            handle_payload_exception(span, payload[:exception_object]) if payload[:exception_object]
+          end
+
+          private
+
+          def tracer
+            Grape::Instrumentation.instance.tracer
+          end
+
+          def span_name(endpoint)
+            "HTTP #{request_method(endpoint)} #{path(endpoint)}"
+          end
+
+          # ActiveSupport::Notifications will attach a `:exception_object` to the payload if there was
+          # an error raised during the execution of the &block associated to the Notification.
+          def handle_payload_exception(span, exception)
+            # Only record exceptions if they were not raised (i.e. do not have a status code in Grape)
+            # or do not have a 5xx status code. These exceptions are recorded by Rack.
+            # See instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb#L155
+            return unless exception.respond_to?('status') && exception.status.to_i < 500
+
+            span.record_exception(exception)
+            span.status = OpenTelemetry::Trace::Status.error("Unhandled exception of type: #{exception.class}")
+          end
+
+          def request_method(endpoint)
+            endpoint.options[:method]&.first
+          end
+
+          def path(endpoint)
+            namespace = endpoint.routes.first.namespace
+            version = endpoint.routes.first.options[:version] || ''
+            prefix = endpoint.routes.first.options[:prefix]&.to_s || ''
```

Comment:
I don't think so, because if the endpoint does not match `endpoint_run` does not run at all. But just to be 100% safe (in the event of a bug in Grape or something like that) I added [this](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/394/commits/5decb4598a2759e1334c7bfe531ef675f340088f#diff-413cd403b56aef0b2f6ecbbc3b77991071e8e77b62c0be98f6bc653e0b6a9356R93), wdyt?